From 46c4a4b4f690572f90b453bb1684ae30a2fc0b6c Mon Sep 17 00:00:00 2001
From: Michael James Gratton <mike@vee.net>
Date: Mon, 21 Mar 2016 11:19:23 +1100
Subject: Workaround for random crash when usign WebKitGTK+ 2.4.10. Fixes Bug
 763933.

* src/client/conversation-viewer/conversation-viewer.vala
  (ConversationViewer::show_images_email): Crash happens when setting
  value for an existing SRC attr on existing IMG element, so remove it
  before setting it.

Signed-off-by: Michael James Gratton <mike@vee.net>
---
 src/client/conversation-viewer/conversation-viewer.vala | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/client/conversation-viewer/conversation-viewer.vala b/src/client/conversation-viewer/conversation-viewer.vala
index 4206857..eb85465 100644
--- a/src/client/conversation-viewer/conversation-viewer.vala
+++ b/src/client/conversation-viewer/conversation-viewer.vala
@@ -1493,8 +1493,11 @@ public class ConversationViewer : Gtk.Box {
                         continue;
                     
                     string src = element.get_attribute("src");
-                    if (!web_view.is_always_loaded(src))
+                    if (!web_view.is_always_loaded(src)) {
+                        // Workaround a WebKitGTK+ 2.4.10 crash. See Bug 763933
+                        element.remove_attribute("src");
                         element.set_attribute("src", web_view.allow_prefix + src);
+                    }
                 }
             }
             
-- 
cgit v0.12

From 45d687569a81058855888a7aed8ff6015deb41c7 Mon Sep 17 00:00:00 2001
From: Adam Dingle <adam@medovina.org>
Date: Thu, 24 Mar 2016 13:52:33 -0400
Subject: Work around crash in WebKitGTK: bug 764152

https://bugzilla.gnome.org/show_bug.cgi?id=764152
---
 src/client/conversation-viewer/conversation-viewer.vala | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/client/conversation-viewer/conversation-viewer.vala b/src/client/conversation-viewer/conversation-viewer.vala
index eb85465..ba807aa 100644
--- a/src/client/conversation-viewer/conversation-viewer.vala
+++ b/src/client/conversation-viewer/conversation-viewer.vala
@@ -2020,6 +2020,7 @@ public class ConversationViewer : Gtk.Box {
                     
                     // Replace the SRC to a data URI, the class to a known label for the popup menu,
                     // and the ALT to its filename, if supplied
+                    img.remove_attribute("src");  // Work around a WebKitGTK+ crash. Bug 764152
                     img.set_attribute("src", assemble_data_uri(mimetype, image_content));
                     img.set_attribute("class", DATA_IMAGE_CLASS);
                     if (!Geary.String.is_empty(filename))
-- 
cgit v0.12


